### PR TITLE
Increase test coverage

### PR DESCRIPTION
**Pull request by kevincheng96**
Add test coverage for events and `preview` functions. 

We also change the `LackAllowance()` custom error to be `InsufficientAllowance()` instead.

### CODE REVIEW

File: test/CometWrapper.t.sol [hunk 1]
```diff
@@ -75,7 +80,185 @@ contract CometWrapperTest is BaseTest, CometMath {
...
+        cometWrapper.deposit(5_000e6, alice);
+        vm.stopPrank();
+
+        assertApproxEqAbs(cometWrapper.underlyingBalance(alice), 5_000e6, 1);
+        skip(14 days);
+        assertGe(cometWrapper.underlyingBalance(alice), 5_000e6);
+    }
+
+    function test_previewDeposit() public {
+        assertEq(cometWrapper.balanceOf(alice), 0e6);
```
Comment (scott-silver) [inline test/CometWrapper.t.sol:97]: nit: `0` instead of `0e6`?

File: test/CometWrapper.t.sol [hunk 2]
```diff
@@ -75,7 +80,185 @@ contract CometWrapperTest is BaseTest, CometMath {
...
+        uint256 bobPreviewedSharesReceived = cometWrapper.previewDeposit(5_000e6);
+        uint256 bobSharesFromAssets = cometWrapper.convertToShares(5_000e6);
+
+        vm.startPrank(bob);
+        comet.allow(wrapperAddress, true);
+        uint256 bobActualSharesReceived = cometWrapper.deposit(5_000e6, bob);
+        vm.stopPrank();
+
+        assertApproxEqAbs(comet.balanceOf(bob), bobCometBalance - 5_000e6, 1);
+        // TODO: investigate rounding
```
Comment (scott-silver) [inline test/CometWrapper.t.sol:125]: is this value different than what you expected?

File: test/CometWrapper.t.sol [hunk 3]
```diff
@@ -138,51 +327,69 @@ contract CometWrapperTest is BaseTest, CometMath {
...
         assertEq(cometWrapper.totalAssets(), comet.balanceOf(wrapperAddress));
 
-        uint256 totalAssets = cometWrapper.maxWithdraw(alice) + cometWrapper.maxWithdraw(bob);
+        uint256 aliceAssets = cometWrapper.maxWithdraw(alice);
+        uint256 bobAssets = cometWrapper.maxWithdraw(bob);
+        uint256 totalAssets = aliceAssets + bobAssets;
         assertLe(totalAssets, cometWrapper.totalAssets());
 
         vm.startPrank(alice);
+        // TODO: investigate!
```
Comment (scott-silver) [inline test/CometWrapper.t.sol:336]: 👀

File: test/CometWrapper.t.sol [hunk 4]
```diff
@@ -255,78 +482,90 @@ contract CometWrapperTest is BaseTest, CometMath {
...
+        assertEq(cometWrapper.allowance(bob, alice), type(uint256).max);
 
         vm.startPrank(alice);
         cometWrapper.transferFrom(bob, alice, 1_000e6);
         assertEq(cometWrapper.allowance(bob, alice), type(uint256).max);
         vm.stopPrank();
     }
 
-    function test__transferFromRevertsOnLackingAllowance() public {
+    function test_transferFrom_revertInsufficientAllowance() public {
```
Comment (scott-silver) [inline test/CometWrapper.t.sol:541]: nit: `test_transferFromRevertInsufficientAllowance` to be consistent with your other names

**Review by scott-silver — APPROVED** [review]
great to see more test coverage 🙌

File: test/CometWrapper.t.sol (hunk 4 above)
Comment (kevincheng96) [reply test/CometWrapper.t.sol]: Updated the other test names to be consistent with this one.

File: test/CometWrapper.t.sol (hunk 2 above)
Comment (kevincheng96) [reply test/CometWrapper.t.sol:125]: I expect there to be some rounding, so this is just a note for me to look into it deeper and make sure the rounding makes sense.

File: test/CometWrapper.t.sol (hunk 1 above)
Comment (kevincheng96) [reply test/CometWrapper.t.sol]: Done.

**Pull request merged**